webpack/scenes/Subscriptions/components/SubscriptionsTable/SubscriptionsTableHelpers.js

```diff
@@ -30,18 +30,40 @@ const buildTableRow = (subscription, availableQuantities, updatedQuantity) => {
       quantity: updatedQuantity[subscription.id],
     };
   }
+
   return baseSubscription;
 };
 
+const buildTableCollapseRow = (subscriptionGroup) => {
+  const first = subscriptionGroup.subscriptions[0];
+  const heading = {};
+
+  heading.id = 0;
+  heading.collapsible = true;
+  heading.contract_number = 'NA';
+  heading.start_date = 'NA';
+  heading.end_date = 'NA';
+  heading.consumed = 'NA';
+  heading.product_id = first.product_id;
+  heading.name = first.name;
+  heading.virt_only = first.virt_only;
+  heading.hypervisor = first.hypervisor;
+
+  return heading;
+};
+
 const buildTableRowsFromGroup = (subscriptionGroup, availableQuantities, updatedQuantity) => {
   const { open, subscriptions } = subscriptionGroup;
+  const rows = [];
 
-  // build row for each subscription
-  if (open) {
-    return subscriptions.map(subscription =>
-      buildTableRow(subscription, availableQuantities, updatedQuantity));
+  if (subscriptions.length > 1) {
+    rows.push(buildTableCollapseRow(subscriptionGroup));
+    if (open) {
+      return rows.concat(subscriptions.map(sub =>
```

Comment:
What about, instead of `return rows.concat` if we were to use `subscriptions.forEach(sub => rows.push(...` ? 

That way, on line 65 you're just returning `rows` no matter what, and it's a little easier to see what's happening.

Reply:
Nice catch.  Thanks for your feedback @jeremylenz .